### DISCUSSION TITLE
WinUI Community Call June 2024 Q&A Follow-up

### DISCUSSION THREAD

**Discussion by duncanmacmichael** [microsoft/microsoft-ui-xaml#9773] [General]
Hi everyone, we'd like to follow up on the June 2024 Community Call here and make sure we address all the questions we couldn't get to during the call.

We appreciate your patience and we want to thank you as always for your great feedback and questions.

## Questions asked in the live comments that were not answered during the call

> NOTE: If there is a question you asked but don't see here, it might be something we're unclear about or don't have a comment for at this time, it might have been a duplicate of another similar question, or it could be just something we missed. Feel free to open a new discussion or raise again during our next community call Q&A.

### Q: \[Multiple questions and comments related to the new TitleBar control\]

**A:** We're really excited at the positive reception of the new TitleBar control that we introduced in 1.6-experimental1 and are actively evaluating feedback to refine the new control. We recommend joining the discussion here in the GitHub repo, either by creating a new discussion post or by joining an existing feedback proposal such as [this one](https://github.com/microsoft/microsoft-ui-xaml/issues/9700).

### Q: Hey if I'm having issues with the store where's the best place to go?

**A:** We recommend submitting feedback through the Feedback Hub if you're having Store issues.

### Q: ​​Are there plans for the CommunityToolkit.Windows to also support wpf? since modernization is going on, and it makes sense for these controls to be in wpf.

**A:** Thanks for the suggestion! We recommend filing this kind of feedback in the [WPF GitHub repo](https://github.com/dotnet/wpf).

### Q: I'm curious if Blend has any future for WinUI

**A:** We're tracking any designer-related feedback with [this](https://github.com/microsoft/microsoft-ui-xaml/issues/5917) GitHub issue. Please feel free to add your feedback to this issue.

### Q: Can we stop using AppWindow in WinUI apps soon and just make Window do these things directly?

**A:** We are looking into improvements in the AppWindow/Window area, so please let us know what features you'd find most valuable to having directly available on Window by filing a feature proposal here in the WinUI repo.

### Q: Will improved Drag and Drop come to general dragging and dropping of elements in WinUI or WinAppSDK. Files can not do any drag and dropping in an Elevated app instance right now

**A:** We don't currently have a timeline for general drag-and-drop support for elevated apps.

### Q: When you tear out a tab is there a way to get to the AppWindow or HWND for the new window?

**A:** Yes! In fact, the app creates the new window as part of handling the `TabView.TabTearOutWindowRequested` event and returns the window to WinUI via the event args.

### Q: Any updates on open sourcing?

**A:** As we mentioned previously in the follow-up to the September 2023 Community Call, we are taking the *source sharing* approach and there is currently no plan for open source. To recap from that answer:

> Q: Will we be able to build from the source or only view due to private dependencies?
> 
> A: To be clear, we are doing shared source, not open source, as there are underlying interwoven components linking WinUI 3 to the system that we can't share. We will not support building, deploying, or running the source code and will not accept PRs for it. We will support viewing, searching, and debugging of the source code when it is available.

### Q: How can you debug and inspect WinUI applications effectively given the absence of browser-based developer tools like those in web development?

**A:** Please add your feedback as a GitHub issue so we can track it as part of developer tooling improvements that we're investigating.

### Q: any plans for other language projections for win app sdk? other than cpp and cs?

**A:** We currently have no plans for extending language projections.

## Next WinUI Community Call

The next WinUI community call will take place on September 11th, 2024. As we approach the call date, we'll post a discussion to gather questions ahead of time and share the call agenda. We look forward to seeing you then!

**Comment by fredemmott** [comment]
> A: To be clear, we are doing shared source, not open source, as there are underlying interwoven components linking WinUI 3 to the system that we can't share. We will not support building, deploying, or running the source code and will not accept PRs for it. We will support viewing, searching, and debugging of the source code when it is available.

This is a big disappointment, and means that WinUI3 will remain unusable in apps that also want to use GPL-licensed libraries. It *clearly* is not covered under the usual exception in either GPL v2:

>  However, as a special exception, the source code distributed need not include anything that is normally distributed (in either source or binary form) with the major components (compiler, kernel, and so on) of the operating system on which the executable runs, unless that component itself accompanies the executable.

... or GPL v3:

> The “System Libraries” of an executable work include anything, other than the work as a whole, that (a) is included in the normal form of packaging a Major Component, but which is not part of that Major Component, and (b) serves only to enable use of the work with that Major Component, or to implement a Standard Interface for which an implementation is available to the public in source code form. A “Major Component”, in this context, means a major essential component (kernel, window system, and so on) of the specific operating system (if any) on which the executable work runs, or a compiler used to produce the work, or an object code interpreter used to run it.

WinUI3 is a just a UI library like any other, often distributed with the app, nearly never with the OS.
